app/src/main/java/org/xbmc/kore/ui/generic/SubtitlesOptionsDialog.java

```diff
@@ -0,0 +1,313 @@
+
+    /**
+     * Handler on which to post RPC callbacks
+     */
+    private Handler callbackHandler = new Handler();
+
+    /**
+     * Activity to communicate potential actions that change what's playing
+     */
+    private HostConnectionObserver hostConnectionObserver;
+
+    /**
+     * The current active player id
+     */
+    private int currentActivePlayerId = -1;
+
+    /**
+     * Default callback for methods that don't return anything
+     */
+    private ApiCallback<String> defaultStringActionCallback = ApiMethod.getDefaultActionCallback();
+
+
+
+    private FragmentActivity dialogActivity;
+
+    public SubtitlesOptionsDialog(Activity activity){
+        hostManager =HostManager.getInstance(activity);
+        hostConnectionObserver=hostManager.getHostConnectionObserver();
+        hostConnectionObserver.registerPlayerObserver(this, true);
+        dialogActivity =(FragmentActivity)activity;
+    }
+
+
+    /**
+     * Display subtitles dialog
+     *
+     */
+    public void Show(){
+        int selectedItem = -1;
+
+        String[] subtitles = new String[(availableSubtitles != null) ?
+                availableSubtitles.size() + ADDED_SUBTITLE_OPTIONS : ADDED_SUBTITLE_OPTIONS];
+
+        subtitles[0] = dialogActivity.getString(R.string.download_subtitle);
+        subtitles[1] = dialogActivity.getString(R.string.subtitle_sync);
+        subtitles[2] = dialogActivity.getString(R.string.none);
+
+        if (availableSubtitles != null) {
+            for (int i = 0; i < availableSubtitles.size(); i++) {
+                PlayerType.Subtitle current = availableSubtitles.get(i);
+                subtitles[i + ADDED_SUBTITLE_OPTIONS] = TextUtils.isEmpty(current.language) ?
+                        current.name : current.language + " | " + current.name;
+                if (current.index == currentSubtitleIndex) {
+                    selectedItem = i + ADDED_SUBTITLE_OPTIONS;
+                }
+            }
+        }
+
+        GenericSelectDialog dialog = GenericSelectDialog.newInstance(this,
+                SELECT_SUBTITLES, dialogActivity.getString(R.string.subtitles), subtitles, selectedItem);
+        dialog.show(dialogActivity.getSupportFragmentManager(), null);
+
+    }
+
+    public void addListener(SubtitlesOptionsDialogListener toAdd) {
+        listeners.add(toAdd);
+    }
+
+    public void RemoteRequired() {
+        // Notify everybody that may be interested.
+        for (SubtitlesOptionsDialogListener hl : listeners)
+            hl.onRemoteRequired();
+    }
+
+
+    /**
+     * Generic dialog select listener
+     * @param token
+     * @param which
+     */
+    public void onDialogSelect(int token, int which) {
+        switch (token) {
+            case SELECT_SUBTITLES:
+                Player.SetSubtitle setSubtitle;
+                // 0 is to download subtitles, 1 is for sync, 2 is for none, other is for a specific subtitle index
+                switch (which) {
+                    case 0:
```

Comment:
Please use class variables (private final static int) for cases to make the code more readable.